### DISCUSSION TITLE
Multiple datasource for archive data

### DISCUSSION THREAD

**Discussion by joshikk** [ControlSystemStudio/phoebus#2006] [Q&A] [answered]
I was trying to connect to remote datasource. So I tried the following in my settings.ini file.
IT does not work.

org.csstudio.trends.databrowser3/urls=jdbc:postgresql://localhost/archive|RDB jdbc:postgresql://MYDATA/archive|RDB
org.csstudio.trends.databrowser3/archives=jdbc:postgresql://localhost/archive|RDB jdbc:postgresql://MYDATA/archive|RDB

org.csstudio.archive/url=jdbc:postgresql://localhost/archive jdbc:postgresql://MYDATA/archive

On using only the second datasource, it is working. Am I making any error?

I see that the configuration for the datasource is at module level. Is there any way in which I can configure for multiple datasources?

Is it possible to have a datasource configuration at IOC or PV level? 
This will help for of distributed datasources.

Regards,
Kuldeep

**Comment by kasemir** [comment, marked as answer]
If all fails, read the manual. Relevant section is https://control-system-studio.readthedocs.io/en/latest/preference_properties.html#trends-databrowser3:
```
# Format:  <url>*<url>|<name>
# List of URLs, separated by '*'.
# Each URL may be followed by an "|alias"
```

If you have multiple data sources, they need to be separated by '*':
```
jdbc:postgresql://host1.mysite.org/archive|PGArchive*jdbc:mysql://host2.mysite.org/archive|MySQLArchive
```

**Comment by joshikk** [comment]
Sorry, I read the preferences listing diagonally
Thank you for your help

Regards,
Kuldeep
